core/src/bundle/usBundleUtils.cpp

```diff
@@ -127,24 +196,82 @@ std::string GetLibraryPath_impl(void *symbol)
   return "";
 }
 
+std::string GetExecutablePath()
+{
+  char pathBuf[1024];	// assuming this is a large enough buffer.
+  if (GetModuleFileNameA(nullptr, pathBuf, sizeof(pathBuf)) == 0 || GetLastError() == ERROR_INSUFFICIENT_BUFFER)
+  {
+    US_DEBUG << "GetModuleFileName failed" << GetLastErrorStr();
+    pathBuf[0] = '\0';
+  }
+  return std::string(pathBuf);
+}
+  
+struct FileInfo {
+  std::string path;
+  DWORD dwVolumeSerialNumber;
+  DWORD nFileIndexHigh;
+  DWORD nFileIndexLow;
+  
+  FileInfo() : path(""), dwVolumeSerialNumber(0), nFileIndexHigh(0), nFileIndexLow(0) {}
```

Comment:
consider removing the default ctor. From the usage of FileInfo, it doesn't appear that a FileInfo object with an empty path is allowed.